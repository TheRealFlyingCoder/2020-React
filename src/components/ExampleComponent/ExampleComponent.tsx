import React from 'react';
//import { useStateValue } from 'StateProvider';
import styles from './ExampleComponent.styles';

const ExampleComponent = (): React.ReactElement => {

    //const [globaleState, setGlobalState] = useStateValue();
    //const [localState, setLocalState] = useState(null);
    
    return (
        <div css={styles}>
            <p className="freshBoi">This is a fresh boi</p>
        </div>
    );    
};

export default ExampleComponent;