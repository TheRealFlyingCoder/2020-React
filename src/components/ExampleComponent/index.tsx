export { default } from './ExampleComponent';
export * from './ExampleComponent';
