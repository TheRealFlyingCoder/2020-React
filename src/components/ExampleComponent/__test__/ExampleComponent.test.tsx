import React from 'react';
import renderer from 'react-test-renderer';
import serializer from 'jest-emotion';
import ExampleComponent from '../ExampleComponent';
//Helps Emotion serialise out the styles for the snapshot
expect.addSnapshotSerializer(serializer)

describe('ExampleComponent', () => {
    test('ExampleComponent renders', () => {
    const component = renderer.create(<ExampleComponent />);
    expect(component.toJSON()).toMatchSnapshot();
    });
});