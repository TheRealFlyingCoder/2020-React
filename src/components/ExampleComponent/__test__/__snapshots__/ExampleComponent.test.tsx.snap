// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent ExampleComponent renders 1`] = `
.emotion-0 .freshBoi {
  color: red;
}

<div
  className="emotion-0"
>
  <p
    className="freshBoi"
  >
    This is a fresh boi
  </p>
</div>
`;
